homeassistant/components/sun.py

```diff
@@ -55,16 +83,33 @@ class Sun(Entity):
 
     entity_id = ENTITY_ID
 
-    def __init__(self, hass, location):
+    def __init__(self, hass, location, config):
         """Initialize the sun."""
+        self._monitored_condtions = config[CONF_MONITORED_CONDITIONS]
+        scan_interval = config.get(CONF_SCAN_INTERVAL)
         self.hass = hass
         self.location = location
-        self._state = self.next_rising = self.next_setting = None
-        self.next_dawn = self.next_dusk = None
-        self.next_midnight = self.next_noon = None
-        self.solar_elevation = self.solar_azimuth = None
-
-        async_track_utc_time_change(hass, self.timer_update, second=30)
+        # Initialize values for attributes to be reported.
+        for attr in self._monitored_condtions:
+            setattr(self, attr, None)
```

Comment:
Please don't do that. Use a internal dict as cache, if you need one. It make all things hard to read and hard to read mean that other make mistakes. Keep it simple

Reply:
Good idea. Thanks! Will do.